fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/impl/object/builder/FabricBlockInternals.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.object.builder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.tag.Tag;
+
+import net.fabricmc.fabric.impl.mining.level.ToolManager;
+
+public final class FabricBlockInternals {
+	private FabricBlockInternals() {
+	}
+
+	public static final Map<Block.Settings, ExtraData> EXTRA_DATA = new HashMap<>();
```

Comment:
Why not mixing the field into the block settings class?

Reply:
Makes sense to reduce Map calls, but that will be to @sfPlayer1 to decide since I sont want to reinvent the entire wheel.

It's already an improvement this isn't a inner class of an api class anymore

Reply:
Block.Settings isn't a builder, this is.

Reply:
Not sure why this replied to the wrong comment. The mobile app sucks. It should be a reply to liach's comment on the material builder.

Reply:
The field should be a lot safer, this asks for the contents to become stale/never read again

Reply:
So to you want me to store the extra data via a mixin in the Block.Settings? In a field